spec/controllers/solutions_controller_spec.rb

```diff
@@ -119,4 +119,52 @@
       end
     end
   end
+
+  describe '#show' do
+    let(:request_url) { "/challenges/#{challenge_id}/solution" }
+    let(:request_headers) { { ACCEPT: 'application/json' } }
+
+    subject do
+      get request_url, headers: request_headers
+      response
+    end
+
+    context 'user is not authenticated' do
+      let(:challenge_id) { 1 }
+      it { expect(subject.status).to eq(401) }
+    end
+
+    context 'user is authenticated and solution exists' do
```

Comment:
You could nest this two context. the main context will be ```user is authenticated``` and inside this context you could tell ```and solution exists``` and ```and solution does not exist```. In this way you can reuse the stuff about the user